Modules/Core/Common/test/itkAnatomicalOrientationGTest.cxx

```diff
@@ -0,0 +1,249 @@
+  d(1, 2) = 1;
+  EXPECT_EQ(OE::LSP, AnatomicalOrientation::DirectionCosinesToOrientation(d));
+
+  d.Fill(0.0);
+  d(1, 0) = 1;
+  d(0, 1) = 1;
+  d(2, 2) = 1;
+  EXPECT_EQ(OE::PLS, AnatomicalOrientation::DirectionCosinesToOrientation(d));
+
+  d.Fill(0.0);
+  d(2, 0) = 1;
+  d(1, 1) = 1;
+  d(0, 2) = 1;
+  EXPECT_EQ(OE::SPL, AnatomicalOrientation::DirectionCosinesToOrientation(d));
+
+  const double                                 data[] = { 0.5986634407395047, 0.22716302314740483, -0.768113953548866,
+                          0.5627936241740271, 0.563067040943212,   0.6051601804419384,
+                          0.5699696670095713, -0.794576911518317,  0.20924175102261847 };
+  ImageType::DirectionType::InternalMatrixType m{ data };
+  d.GetVnlMatrix() = m;
+  EXPECT_EQ(OE::PIR, AnatomicalOrientation::DirectionCosinesToOrientation(d));
+}
+
+TEST(AnatomicalOrientation, OrientationToDirectionCosines)
+{
+  using itk::AnatomicalOrientation;
+  using ImageType = itk::Image<float, 3>;
+  using OE = AnatomicalOrientation::ToEnum;
+
+  ImageType::DirectionType d;
+  d.SetIdentity();
+
+  EXPECT_EQ(d, AnatomicalOrientation::OrientationToDirectionCosines(OE::LPS));
+
+  d.Fill(0.0);
+  d(0, 0) = -1.0;
+  d(1, 1) = -1.0;
+  d(2, 2) = -1.0;
+  EXPECT_EQ(d, AnatomicalOrientation::OrientationToDirectionCosines(OE::RAI));
+
+  d.Fill(0.0);
+  d(2, 0) = 1;
+  d(0, 1) = 1;
+  d(1, 2) = 1;
+  EXPECT_EQ(d, AnatomicalOrientation::OrientationToDirectionCosines(OE::SLP));
+
+  d.Fill(0.0);
+  d(1, 0) = 1;
+  d(2, 1) = 1;
+  d(0, 2) = 1;
+  EXPECT_EQ(d, AnatomicalOrientation::OrientationToDirectionCosines(OE::PSL));
+
+  d.Fill(0.0);
+  d(0, 0) = 1;
+  d(2, 1) = 1;
+  d(1, 2) = 1;
+  EXPECT_EQ(d, AnatomicalOrientation::OrientationToDirectionCosines(OE::LSP));
+
+  d.Fill(0.0);
+  d(1, 0) = 1;
+  d(0, 1) = 1;
+  d(2, 2) = 1;
+  EXPECT_EQ(d, AnatomicalOrientation::OrientationToDirectionCosines(OE::PLS));
+
+  d.Fill(0.0);
+  d(2, 0) = 1;
+  d(1, 1) = 1;
+  d(0, 2) = 1;
+  EXPECT_EQ(d, AnatomicalOrientation::OrientationToDirectionCosines(OE::SPL));
+}
+
+TEST(AntomicalOrientation, ToFromEnumInteroperability)
+{
+
+  using OE = itk::AnatomicalOrientation::ToEnum;
+  using FromOE = itk::AnatomicalOrientation::FromEnum;
+  using CE = itk::AnatomicalOrientation::CoordinateEnum;
+
+  static_assert(int(OE::RAI) == int(FromOE::LPS));
+  static_assert(int(OE::LPS) == int(FromOE::RAI));
+  static_assert(int(OE::RAS) == int(FromOE::LPI));
+  static_assert(int(OE::LPI) == int(FromOE::RAS));
+  static_assert(int(OE::PIR) == int(FromOE::ASL));
+  static_assert(int(OE::ASL) == int(FromOE::PIR));
+
+  itk::AnatomicalOrientation itk_rai(FromOE::RAI);
+
+  EXPECT_EQ(itk_rai, itk::AnatomicalOrientation(OE::LPS));
+  EXPECT_EQ(itk_rai.GetAsOrientation(), OE::LPS);
+  EXPECT_EQ(itk_rai.GetAsString(), "LPS");
+  const std::array<CE, 3> expected_terms = { { CE::LeftToRight, CE::AnteriorToPosterior, CE::InferiorToSuperior } };
+  EXPECT_EQ(itk_rai.GetTerms(), expected_terms);
```

Comment:
The expected result is in correctly defined here. The following is the output from the [test failure](https://open.cdash.org/tests/1681790452):

/Users/runner/work/1/s/Modules/Core/Common/test/itkAnatomicalOrientationGTest.cxx:227: Failure
Expected equality of these values:
  itk_rai.GetTerms()
    Which is: { right-to-left, anterior-to-posterior, inferior-to-superior }
  expected_terms
    Which is: { left-to-right, anterior-to-posterior, inferior-to-superior }